Sources/AppcuesKit/Presentation/Traits/Appcues/ExperienceWrapperSlideAnimator.swift

```diff
@@ -0,0 +1,115 @@
+        self.edgeIn = edgeIn
+        self.edgeOut = edgeOut
+    }
+
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+
+        let animator: UIViewPropertyAnimator
+        switch transitionType {
+        case .presentation:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, dampingRatio: 0.82)
+        case .dismissal:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, curve: UIView.AnimationCurve.easeIn)
+        }
+
+        switch transitionType {
+        case .presentation:
+            // We need to add the modal to the view hierarchy,
+            // and perform the animation.
+            if let toView = transitionContext.view(forKey: .to) {
+                toView.translatesAutoresizingMaskIntoConstraints = false
+                transitionContext.containerView.addSubview(toView)
+                toView.pin(to: transitionContext.containerView)
+                toView.layoutIfNeeded()
+            }
+
+            // move the modal off screen
+            UIView.performWithoutAnimation { beginTransition(transitionContext) }
+            // then slide it in
+            animator.addAnimations { [weak self] in self?.endTransition() }
+        case .dismissal:
+            // The modal is already in the view hierarchy,
+            // so we just perform the animation.
+            animator.addAnimations { [weak self] in self?.beginTransition(transitionContext) }
+        }
+
+        animator.addCompletion { _ in
+            transitionContext.completeTransition(true)
+        }
+
+        animator.startAnimation()
+    }
+
+    private func beginTransition(_ transitionContext: UIViewControllerContextTransitioning) {
+        var offsetX: CGFloat
+        var offsetY: CGFloat
+
+        let edge = transitionType == .presentation ? edgeIn : edgeOut
+
+        switch edge {
+        case .top:
+            offsetX = 0
+            offsetY = -1 * view.shadowWrappingView.frame.maxY
+        case .leading:
+            offsetX = -1 * view.shadowWrappingView.frame.maxX
+            offsetY = 0
+        case .bottom:
+            offsetX = 0
+            offsetY = transitionContext.containerView.bounds.height - view.shadowWrappingView.frame.minY
+        case .trailing:
+            offsetX = transitionContext.containerView.bounds.width - view.shadowWrappingView.frame.minX
+            offsetY = 0
+        case .center:
+            // center+center special case - will be similar to bottom, but not start fully offscreen
+            offsetX = 0
+            offsetY = view.shadowWrappingView.frame.height / 2.0
+            view.contentWrapperView.alpha = 0.0
+        }
+
+        view.contentWrapperView.transform = CGAffineTransform.identity.translatedBy(x: offsetX, y: offsetY)
```

Comment:
this is where it sets up some initial translation on the contentWrapperView, depending on the desired entrance edge, which is then animated back to the expected spot in the end.